src/services/inplayer.checkout.service.ts

```diff
@@ -0,0 +1,232 @@
+import InPlayer, { GetAccessFee, MerchantPaymentMethod } from '@inplayer-org/inplayer.js';
+
+import type {
+  CardPaymentData,
+  CreateOrder,
+  CreateOrderArgs,
+  GetEntitlements,
+  GetEntitlementsResponse,
+  GetOffers,
+  GetPaymentMethods,
+  Offer,
+  Order,
+  Payment,
+  PaymentMethod,
+  PaymentWithAdyen,
+  PaymentWithoutDetails,
+  PaymentWithPayPal,
+  UpdateOrder,
+} from '#types/checkout';
+
+export const createOrder: CreateOrder = async (payload) => {
+  return {
+    errors: [],
+    responseData: {
+      message: '',
+      order: processOrder(payload),
+      success: true,
+    },
+  };
+};
+
+export const getOffers: GetOffers = async (payload) => {
+  try {
+    const assetId = parseInt(`${payload?.offerIds?.[0]}`);
+    const { data } = await InPlayer.Asset.getAssetAccessFees(assetId);
+
+    // @ts-ignore
+    //TODO fix this type in the InPlayer SDK
+    return data?.map((offer: GetAccessFee) => processOffer(offer)) as Offer[];
+  } catch {
+    throw new Error('Failed to get offers');
+  }
+};
+
+export const getPaymentMethods: GetPaymentMethods = async () => {
+  try {
+    const response = await InPlayer.Payment.getPaymentMethods();
+    const paymentMethods: PaymentMethod[] = [];
+    response.data.forEach((method: MerchantPaymentMethod) => {
+      if (['card', 'paypal'].includes(method.method_name.toLowerCase())) {
+        paymentMethods.push(processPaymentMethod(method));
+      }
+    });
+    return {
+      errors: [],
+      responseData: {
+        message: '',
+        paymentMethods,
+        status: 1,
+      },
+    };
+  } catch {
+    throw new Error('Failed to get payment methods');
```

Comment:
It might be nice on all of these errors to keep the original error as the `cause`

Reply:
That may require some effort since the sdk uses axios, and not all of our endpoints have the same error response data unfortunately.. We will need to create a handler and go through all the endpoints to format the errors. Not sure if it's worth the effort.

Reply:
I mean something like this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#rethrowing_an_error_with_a_cause

Reply:
aham, to include the entire error object as the cause, but not only the message. That's okay then.